src/lib/redux/customerDataSlice.ts

```diff
@@ -14,31 +63,34 @@ export const EMPTY_CUSTOMER: CustomerRecord = {
     hasmeter: false,
     outstandingBalance: '',
     meterReadingIds: [],
-    meterReadings: [],
     paymentIds: [],
-    payments: [],
     customerUpdateIds: [],
     customerUpdates: [],
     totalAmountBilledfromInvoices: 0,
     totalAmountPaidfromPayments: 0,
 }
 
-const customersAdapter = createEntityAdapter<CustomerRecord>();
+export const EMPTY_PAYMENT: PaymentRecord = {
+    id: '',
+    amount: 0,
+    date: '',
+    billedToId: '',
+    collectedById: '',
+}
 
-// Returns customers in the context of the current site
-export const {
-    selectEntities: selectAllCustomers,
-    selectAll: selectAllCustomersArray,
-    selectById: selectCustomerById,
-    selectIds: selectCustomerIds
-} = customersAdapter.getSelectors(
-    (state: RootState) => state.customerData.sitesCustomers[state.siteData.currentSite.id]);
+export const EMPTY_METER_READING: MeterReadingRecord = {
+    id: '',
+    reading: 0,
+    amountBilled: 0,
+    date: '',
+    meterNumber: -1,
+    customerId: '',
+}
 
-// TODO: @julianrkung, We should eventually make this a Record<SiteId, Record<CustomerId, CustomerRecord>> map for efficiency
-// We can't do this yet because customers created while offline do not receive an id yet
-interface customerDataSliceState {
-    sitesCustomers: Record<SiteId, EntityState<CustomerRecord>>;
-    currentCustomerId: string;
+export const EMPTY_CUSTOMER_DATA: CustomerData = {
```

Comment:
This could be confusing with the existing `EMPTY_CUSTOMER`, so it may make more sense to rename this to `EMPTY_SITE_CUSTOMER_DATA` (at least I had it called `EMPTY_SITE_INVENTORY_DATA` in inventory) to emphasize that represents all customer data for a site.